phoenix-core/src/it/java/org/apache/phoenix/end2end/TransformToolIT.java

```diff
@@ -413,59 +432,314 @@ public void testTransformMutationFailureRepair() throws Exception {
-            assertEquals(1, getRowCount(conn,dataTableFullName));
+            assertEquals(2, getRowCount(conn, record.getNewPhysicalTableName()));
+            assertEquals(2, getRowCount(conn,dataTableFullName));
 
             IndexRegionObserver.setFailPostIndexUpdatesForTesting(false);
             IndexRegionObserver.setFailDataTableUpdatesForTesting(true);
             try {
-                IndexToolIT.upsertRow(stmt1, 3);
+                IndexToolIT.upsertRow(stmt1, 4);
                 fail("Data table upsert should have failed");
             } catch (Exception e) {
             }
-            assertEquals(2, getRowCount(conn, record.getNewPhysicalTableName()));
-            assertEquals(1, getRowCount(conn,dataTableFullName));
+            try {
+                IndexToolIT.upsertRow(stmt1, 5);
+                fail("Data table upsert should have failed");
+            } catch (Exception e) {
+            }
+            assertEquals(4, getRowCount(conn, record.getNewPhysicalTableName()));
+            assertEquals(2, getRowCount(conn,dataTableFullName));
 
             IndexRegionObserver.setFailDataTableUpdatesForTesting(false);
 
             List<String> args = getArgList(schemaName, dataTableName, null,
-                    null, null, null, false, false, false, false);
+                    null, null, null, false, false, false, false, true);
             runTransformTool(args.toArray(new String[0]), 0);
 
-            //TODO: Implement GlobalIndexChecker like repair for unverified rows
-            assertEquals(1, getRowCount(conn.unwrap(PhoenixConnection.class).getQueryServices()
+            assertEquals(2, getRowCount(conn.unwrap(PhoenixConnection.class).getQueryServices()
                     .getTable(Bytes.toBytes(dataTableFullName)), false));
-//            assertEquals(1, getRowCount(conn.unwrap(PhoenixConnection.class).getQueryServices()
-//                    .getTable(Bytes.toBytes(record.getNewPhysicalTableName())), false));
+            assertEquals(0, getRowCountForEmptyColValue(conn, record.getNewPhysicalTableName(), UNVERIFIED_BYTES));
+            assertEquals(2, getRowCountForEmptyColValue(conn, record.getNewPhysicalTableName(), VERIFIED_BYTES));
+        } finally {
+            IndexRegionObserver.setFailPreIndexUpdatesForTesting(false);
+            IndexRegionObserver.setFailDataTableUpdatesForTesting(false);
+            IndexRegionObserver.setFailPostIndexUpdatesForTesting(false);
+        }
+    }
+
+    @Test
+    public void testTransformVerifiedForTransactionalTable() throws Exception {
+        // TODO: Column encoding is not supported for OMID. Have omid test for other type of transforms.
+        // For now, we don't support transforms other than storage and column encoding type change.
+        //testVerifiedForTransactionalTable("OMID");
+        testVerifiedForTransactionalTable("TEPHRA");
+    }
+
+    private void testVerifiedForTransactionalTable(String provider) throws Exception{
```

Comment:
@stoty added this test for TEPHRA. For OMID, we cannot have transform for now because the only type we support is changing column encoding and storage format encoding which is not compatible with OMID. When we have different type of transforms, we can add a test for OMID too.

Reply:
@gokceni -  I see existing error messages UNSUPPORTED_COLUMN_ENCODING_FOR_TXN_PROVIDER and UNSUPPORTED_STORAGE_FORMAT_FOR_TXN_PROVIDER -- do those get thrown on an ALTER of an OMID table or just a CREATE? If just a CREATE, it should probably do so for ALTER too. (Can be a separate JIRA.)